students/jim_krecek/lesson04/mailroom_p2.py

```diff
@@ -0,0 +1,60 @@
+
+data = {"Nate Secinaro":[12,44,9],"Jess Reid":[3],"Pat Carrier":[65,41],"Zac Fisher":[13,4,4],"Jim Krecek":[3,33,57]}
+
+prompt = "\nWhat do you want to do?\n -1 Send a Thank You\n -2 Create a Report\n -3 Send letters to all donors\n -4 Quit\n>>> "
+
+def send_ty():
+    askname = input("\nEnter the full name of the donor or type 'list' to see the list of donors:\n>>> ")
+    if askname.lower() == "list":
+        print("\nHere is a list of all of the donors:")
+        for d in data.keys():
+            print(d)
+    else:
+        for d in data.keys():
+            if askname.lower() == d.lower():
+                askmoney = input("\nHow much did this person donate?\n>>> ")
+                data[d].append(int(askmoney))
+                emailtext = "\nDear {},\n\nThank you for your generous donation of ${}.\n\nSincerely, Jim".format(d,askmoney)
+                print(emailtext)
+                break
```

Comment:
there should be no looping over dictionary - the best feature of dictionary is access by key.

your code should look something like:
```
if data.get(askname):
    # code here
else:
    # code here
```